test/Performance/run-perftests.py

```diff
@@ -0,0 +1,238 @@
+#!/usr/bin/env python
+
+import argparse
+import glob
+import os
+import re
+import subprocess
+import sys
+
+SCRIPT_ROOT_PATH = os.path.dirname(os.path.realpath(__file__))
+PERFTEST_JSON_PATH = os.path.join(SCRIPT_ROOT_PATH, 'project.json')
+XUNITPERF_REPO_URL = 'https://github.com/microsoft/xunit-performance.git'
+
+script_args = None
+
+class FatalError(Exception):
+    def __init__(self, message):
+        self.message = message
+
+def check_requirements():
+    try:
+        run_command('git', '--version', quiet = True)
+    except:
+        raise FatalError("git not found, please make sure that it's installed and on path.")
+
+    try:
+        run_command('msbuild', '-version', quiet = True)
+    except:
+        raise FatalError("msbuild not found, please make sure that it's installed and on path.")
+
+    if script_args.xunit_perf_path == None:
+        raise FatalError("Don't know where to clone xunit-performance. Please specify --xunit-perf-path <path>. " +
+            "You can also set/export XUNIT_PERFORMANCE_PATH to not have to set the value every time.")
+
+def process_arguments():
+    parser = argparse.ArgumentParser(
+        description = "Runs CLI perf tests. Requires 'git' and 'msbuild' to be on the PATH.",
+    )
+
+    parser.add_argument(
+        'test_cli',
+        help = "full path to the dotnet.exe under test",
+    )
+    parser.add_argument(
+        '--runid', '--name', '-n',
+        help = "unique ID for this run",
+        required = True,
+    )
+    parser.add_argument(
+        '--base', '--baseline', '-b',
+        help = "full path to the baseline dotnet.exe",
+        metavar = 'baseline_cli',
+        dest = 'base_cli',
+    )
+    parser.add_argument(
+        '--xunit-perf-path', '-x',
+        help = """Path to local copy of the xunit-performance repository.
+            Required unless the environment variable XUNIT_PERFORMANCE_PATH is defined.""",
+        default = os.environ.get('XUNIT_PERFORMANCE_PATH'),
+        metavar = 'path',
+    )
+    parser.add_argument(
+        '--rebuild', '--rebuild-tools', '-r',
+        help = "Rebuilds the test tools from scratch.",
+        action = 'store_true',
+    )
+    parser.add_argument(
+        '--verbose', '-v',
+        help = "Shows the output of all commands run by this script",
+        action = 'store_true',
+    )
+
+    global script_args
+    script_args = parser.parse_args()
+
+def run_command(*vargs, **kwargs):
+    title = kwargs['title'] if 'title' in kwargs else None
+    from_dir = kwargs['from_dir'] if 'from_dir' in kwargs else None
+    quiet = kwargs['quiet'] if 'quiet' in kwargs else False
+
+    quoted_args = map(lambda x: '"{x}"'.format(x=x) if ' ' in x else x, vargs)
```

Comment:
quoting the args like this won't reflect the same escaping performed in the internals of subprocess.call, 

So the printed cmd_line may not repro the same issues, is that alright?